tree/treeplayer/inc/ROOT/TVec.hxx

```diff
@@ -0,0 +1,536 @@
+   TVec<decltype(f(v0[0], v1[1]))> w(v0.size());
+   std::transform(v0.begin(), v0.end(), v1.begin(), w.begin(), std::forward<F>(f));
+   return w;
+}
+
+template <typename T, typename V, typename F>
+TVec<T> &OperateInPlace(TVec<T> &v0, const TVec<V> &v1, std::string_view opName, F &&f)
+{
+   const auto v0size = v0.size();
+   CheckSizes(v0size, v1.size(), opName);
+   std::transform(v0.begin(), v0.end(), v1.begin(), v0.begin(), std::forward<F>(f));
+   return v0;
+}
+
+template <typename T, typename F>
+auto Operate(const TVec<T> &v, F &&f) -> TVec<decltype(f(v[0]))>
+{
+   TVec<decltype(f(v[0]))> w(v.size());
+   std::transform(v.begin(), v.end(), w.begin(), std::forward<F>(f));
+   return w;
+}
+
+template <typename T, typename F>
+TVec<T> &OperateInPlace(TVec<T> &v, F &&f)
+{
+   std::transform(v.begin(), v.end(), v.begin(), std::forward<F>(f));
+   return v;
+}
+
+template <typename...>
+struct TIsOneOf {
+   static constexpr bool value = false;
+};
+
+template <typename F, typename S, typename... T>
+struct TIsOneOf<F, S, T...> {
+   static constexpr bool value = std::is_same<F, S>::value || TIsOneOf<F, T...>::value;
+};
+
+template <typename T>
+struct TIsChar {
+   static constexpr bool value =
+      TIsOneOf<typename std::decay<T>::type, char, signed char, unsigned char, wchar_t, char16_t, char32_t>::value;
+};
+
+} // End of VecOps NS
+
+} // End of Internal NS
+
+namespace Detail {
+
+namespace VecOps {
+
+template <typename>
+struct TIsTVec {
+   static const bool value = false;
+};
+
+template <typename T>
+struct TIsTVec<ROOT::Experimental::VecOps::TVec<T>> {
+   static const bool value = true;
+};
+
+} // End of VecOps NS
+
+} // End of Detail NS
+
+namespace Experimental {
+
+namespace VecOps {
+
+template <typename T>
+class TVec {
+private:
+   ROOT::Detail::VecOps::TVecImpl<T> fData;
+
+public:
+   using Impl_t = typename ROOT::Detail::VecOps::TVecImpl<T>;
+   using value_type = typename Impl_t::value_type;
+   using size_type = typename Impl_t::size_type;
+   using difference_type = typename Impl_t::difference_type;
+   using reference = typename Impl_t::reference;
+   using const_reference = typename Impl_t::const_reference;
+   using pointer = typename Impl_t::pointer;
+   using const_pointer = typename Impl_t::const_pointer;
+   using iterator = typename Impl_t::iterator;
+   using const_iterator = typename Impl_t::const_iterator;
+   using reverse_iterator = typename Impl_t::reverse_iterator;
+
+   // ctors
+   TVec() {}
+   TVec(size_type count, const T &value) : fData(count, value) {}
+   explicit TVec(size_type count) : fData(count) {}
+   TVec(const std::vector<T> &other) { std::copy(other.begin(), other.end(), fData.begin()); }
+   TVec(std::initializer_list<T> init) : fData(init) {}
+   TVec(pointer p, size_type n) : fData(n, T(), ROOT::Detail::VecOps::TAdoptAllocator<T>(p, n)) {}
+   // assignment
+   TVec<T> &operator=(std::initializer_list<T> ilist) { return fData = ilist; }
```

Comment:
`operator=(const TVec&)` and `operator=(TVec &&)` are missing

Reply:
I left all of the package intentionally aside to have them defaulted: the only data member is an stl vector. Do you see a particular need for implementing them?

Reply:
It would be better to explicitly declare them as `default` to make sure that they will not be silently deleted by future changes to the library. Move-assignment is particularly tricky, because if deleted C++ defaults to use the copy-assignment, with silent loss in performance (hard to catch even in unit tests).

Reply:
sorry. See above :(